BUG: 3 failing tests with scipy 1.17.1 on Arm `neoverse_v1` (AWS Graviton3)

### Describe your issue.

I'm hitting a couple of test failures when installing scipy 1.17.1 from source.

Naively I would say these are negligible floating-point inaccuracies, but I don't know if they actual signal a real problem or not (either with the installation, with the code, or with the tests themselves).

Any feedback on this is very welcome, happy to provide more details if needed.

### Reproducing Code Example

```python
Run the scipy test suite with `python -m pytest -v scipy`
```

### Error message

```shell
__________________________________________________________________________ TestWorkers.test_smoke[trust-constr-2] __________________________________________________________________________

self = <scipy.optimize.tests.test_optimize.TestWorkers object at 0x40009fab8750>, workers = 2, method = 'trust-constr'

    def test_smoke(self, workers, method):
        # checks parallelised optimization output is same as serial
        workers = workers or map

        kwds = {'jac': None, 'hess': None}
        if method in ['Newton-CG', 'trust-ncg', 'trust-krylov']:
            #  methods that require a callable jac
            kwds['jac'] = rosen_der
            kwds['hess'] = '2-point'

        with MapWrapper(workers) as mf:
            res = optimize.minimize(
                rosen, self.x0, options={"workers":mf}, method=method, **kwds
            )
        res_default = optimize.minimize(
            rosen, self.x0, method=method, **kwds
        )
>       assert_equal(res.x, res_default.x)
E       AssertionError:
E       Arrays are not equal
E
E       Mismatched elements: 3 / 3 (100%)
E       Mismatch at indices:
E        [0]: 0.9999973128044448 (ACTUAL), 0.9999973128044447 (DESIRED)
E        [1]: 0.9999946271034812 (ACTUAL), 0.9999946271034814 (DESIRED)
E        [2]: 0.9999892467941509 (ACTUAL), 0.9999892467941512 (DESIRED)
E       Max absolute difference among violations: 3.33066907e-16
E       Max relative difference among violations: 3.33070489e-16
E        ACTUAL: array([0.999997, 0.999995, 0.999989])
E        DESIRED: array([0.999997, 0.999995, 0.999989])

scipy/optimize/tests/test_optimize.py:3469: AssertionError

__________________________________________________________________________ TestPdist.test_pdist_correlation_iris ___________________________________________________________________________

self = <scipy.spatial.tests.test_distance.TestPdist object at 0x4000aebb3f50>

    @pytest.mark.slow
    def test_pdist_correlation_iris(self):
        eps = 1e-7
        X = eo['iris']
        Y_right = eo['pdist-correlation-iris']
        Y_test1 = wpdist(X, 'correlation')
>       assert_allclose(Y_test1, Y_right, rtol=eps)
E       AssertionError:
E       Not equal to tolerance rtol=1e-07, atol=0
E
E       Mismatched elements: 1 / 11175 (0.00895%)
E       Mismatch at index:
E        [10039]: 2.220446049250313e-16 (ACTUAL), 0.0 (DESIRED)
E       Max absolute difference among violations: 2.22044605e-16
E       Max relative difference among violations: inf
E        ACTUAL: array([4.001339e-03, 2.608895e-05, 1.831548e-03, ..., 4.291874e-03,
E              5.539879e-03, 8.412290e-05], shape=(11175,))
E        DESIRED: array([4.001339e-03, 2.608895e-05, 1.831548e-03, ..., 4.291874e-03,
E              5.539879e-03, 8.412290e-05], shape=(11175,))

scipy/spatial/tests/test_distance.py:996: AssertionError

______________________________________________________________________ TestPdist.test_pdist_correlation_iris_float32 _______________________________________________________________________

self = <scipy.spatial.tests.test_distance.TestPdist object at 0x4000aebb3fb0>

    @pytest.mark.slow
    def test_pdist_correlation_iris_float32(self):
        eps = 1e-7
        X = eo['iris']
        Y_right = np.float32(eo['pdist-correlation-iris'])
        Y_test1 = wpdist(X, 'correlation')
>       assert_allclose(Y_test1, Y_right, rtol=eps, verbose=verbose > 2)
E       AssertionError:
E       Not equal to tolerance rtol=1e-07, atol=0
E
E       Mismatched elements: 1 / 11175 (0.00895%)
E       Mismatch at index:
E        [10039]: 2.220446049250313e-16 (ACTUAL), 0.0 (DESIRED)
E       Max absolute difference among violations: 2.22044605e-16
E       Max relative difference among violations: inf

scipy/spatial/tests/test_distance.py:1004: AssertionError
```

### SciPy/NumPy/Python version and system information

```shell
- GCC 15.2.0 as compiler (with `-O1 -mcpu=native -fno-math-errno -fPIC` as compiler options)
- Python 3.14.2 (self-installed from source)
- BLAS/LAPACK library: FlexiBLAS 3.5.0 as frontend for OpenBLAS 0.3.32
- numpy 2.4.6
- configuration with Meson 1.10.2:

meson setup --prefix ... --optimization=1   -Dblas=flexiblas  -Dlapack=flexiblas  -Dlibdir=lib  -Db_ndebug=true /tmp/scipy-1.17.1


output of `scipy.show_config()`:

1.17.1 2.4.6 sys.version_info(major=3, minor=14, micro=2, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    has ilp64: false
    include directory: /tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include/flexiblas
    lib directory: /tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib
    name: flexiblas
    openblas configuration: unknown
    pc file directory: /tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib/pkgconfig
    version: 3.5.0
  lapack:
    detection method: pkgconfig
    found: true
    has ilp64: false
    include directory: /tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include/flexiblas
    lib directory: /tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib
    name: flexiblas
    openblas configuration: unknown
    pc file directory: /tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib/pkgconfig
    version: 3.5.0
  pybind11:
    detection method: pkgconfig
    include directory: /include
    name: pybind11
    version: 3.0.4
Compilers:
  c:
    args: -O1, -mcpu=native, -fno-math-errno, -fPIC, -I/tmp/software/FFTW/3.3.10-GCC-15.2.0/include,
      -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include/flexiblas, -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include,
      -I/tmp/software/Python/3.14.2-GCCcore-15.2.0/include, -I/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/include,
      -I/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/include
    commands: gcc
    linker: ld.bfd
    linker args: -L/tmp/software/FFTW/3.3.10-GCC-15.2.0/lib64, -L/tmp/software/FFTW/3.3.10-GCC-15.2.0/lib,
      -L/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib64, -L/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib,
      -L/tmp/software/GCCcore/15.2.0/lib64, -L/tmp/software/GCCcore/15.2.0/lib,
      -L/tmp/software/Python-bundle-PyPI/2026.04-GCCcore-15.2.0/lib64, -L/tmp/software/Python-bundle-PyPI/2026.04-GCCcore-15.2.0/lib,
      -L/tmp/software/Python/3.14.2-GCCcore-15.2.0/lib64, -L/tmp/software/Python/3.14.2-GCCcore-15.2.0/lib,
      -L/tmp/software/spin/0.18-GCCcore-15.2.0/lib64, -L/tmp/software/spin/0.18-GCCcore-15.2.0/lib,
      -L/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/lib64, -L/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/lib,
      -L/tmp/software/Cython/3.2.4-GCCcore-15.2.0/lib64, -L/tmp/software/Cython/3.2.4-GCCcore-15.2.0/lib,
      -L/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/lib64, -L/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/lib,
      -L/tmp/software/meson-python/0.19.0-GCCcore-15.2.0/lib64, -L/tmp/software/meson-python/0.19.0-GCCcore-15.2.0/lib,
      -L/tmp/software/Meson/1.10.2-GCCcore-15.2.0/lib64, -L/tmp/software/Meson/1.10.2-GCCcore-15.2.0/lib,
      -L/tmp/software/hypothesis/6.153.2-GCCcore-15.2.0/lib64, -L/tmp/software/hypothesis/6.153.2-GCCcore-15.2.0/lib,
      -O1, -mcpu=native, -fno-math-errno, -fPIC, -I/tmp/software/FFTW/3.3.10-GCC-15.2.0/include,
      -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include/flexiblas, -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include,
      -I/tmp/software/Python/3.14.2-GCCcore-15.2.0/include, -I/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/include,
      -I/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/include
    name: gcc
    version: 15.2.0
  c++:
    args: -O1, -mcpu=native, -fno-math-errno, -fPIC, -I/tmp/software/FFTW/3.3.10-GCC-15.2.0/include,
      -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include/flexiblas, -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include,
      -I/tmp/software/Python/3.14.2-GCCcore-15.2.0/include, -I/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/include,
      -I/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/include
    commands: g++
    linker: ld.bfd
    linker args: -L/tmp/software/FFTW/3.3.10-GCC-15.2.0/lib64, -L/tmp/software/FFTW/3.3.10-GCC-15.2.0/lib,
      -L/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib64, -L/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib,
      -L/tmp/software/GCCcore/15.2.0/lib64, -L/tmp/software/GCCcore/15.2.0/lib,
      -L/tmp/software/Python-bundle-PyPI/2026.04-GCCcore-15.2.0/lib64, -L/tmp/software/Python-bundle-PyPI/2026.04-GCCcore-15.2.0/lib,
      -L/tmp/software/Python/3.14.2-GCCcore-15.2.0/lib64, -L/tmp/software/Python/3.14.2-GCCcore-15.2.0/lib,
      -L/tmp/software/spin/0.18-GCCcore-15.2.0/lib64, -L/tmp/software/spin/0.18-GCCcore-15.2.0/lib,
      -L/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/lib64, -L/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/lib,
      -L/tmp/software/Cython/3.2.4-GCCcore-15.2.0/lib64, -L/tmp/software/Cython/3.2.4-GCCcore-15.2.0/lib,
      -L/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/lib64, -L/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/lib,
      -L/tmp/software/meson-python/0.19.0-GCCcore-15.2.0/lib64, -L/tmp/software/meson-python/0.19.0-GCCcore-15.2.0/lib,
      -L/tmp/software/Meson/1.10.2-GCCcore-15.2.0/lib64, -L/tmp/software/Meson/1.10.2-GCCcore-15.2.0/lib,
      -L/tmp/software/hypothesis/6.153.2-GCCcore-15.2.0/lib64, -L/tmp/software/hypothesis/6.153.2-GCCcore-15.2.0/lib,
      -O1, -mcpu=native, -fno-math-errno, -fPIC, -I/tmp/software/FFTW/3.3.10-GCC-15.2.0/include,
      -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include/flexiblas, -I/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/include,
      -I/tmp/software/Python/3.14.2-GCCcore-15.2.0/include, -I/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/include,
      -I/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/include
    name: gcc
    version: 15.2.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.2.4
  fortran:
    args: -O1, -mcpu=native, -fno-math-errno, -fPIC
    commands: gfortran
    linker: ld.bfd
    linker args: -L/tmp/software/FFTW/3.3.10-GCC-15.2.0/lib64, -L/tmp/software/FFTW/3.3.10-GCC-15.2.0/lib,
      -L/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib64, -L/tmp/software/FlexiBLAS/3.5.0-GCC-15.2.0/lib,
      -L/tmp/software/GCCcore/15.2.0/lib64, -L/tmp/software/GCCcore/15.2.0/lib,
      -L/tmp/software/Python-bundle-PyPI/2026.04-GCCcore-15.2.0/lib64, -L/tmp/software/Python-bundle-PyPI/2026.04-GCCcore-15.2.0/lib,
      -L/tmp/software/Python/3.14.2-GCCcore-15.2.0/lib64, -L/tmp/software/Python/3.14.2-GCCcore-15.2.0/lib,
      -L/tmp/software/spin/0.18-GCCcore-15.2.0/lib64, -L/tmp/software/spin/0.18-GCCcore-15.2.0/lib,
      -L/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/lib64, -L/tmp/software/pybind11/3.0.4-GCCcore-15.2.0/lib,
      -L/tmp/software/Cython/3.2.4-GCCcore-15.2.0/lib64, -L/tmp/software/Cython/3.2.4-GCCcore-15.2.0/lib,
      -L/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/lib64, -L/tmp/software/pkgconf/2.5.1-GCCcore-15.2.0/lib,
      -L/tmp/software/meson-python/0.19.0-GCCcore-15.2.0/lib64, -L/tmp/software/meson-python/0.19.0-GCCcore-15.2.0/lib,
      -L/tmp/software/Meson/1.10.2-GCCcore-15.2.0/lib64, -L/tmp/software/Meson/1.10.2-GCCcore-15.2.0/lib,
      -L/tmp/software/hypothesis/6.153.2-GCCcore-15.2.0/lib64, -L/tmp/software/hypothesis/6.153.2-GCCcore-15.2.0/lib,
      -O1, -mcpu=native, -fno-math-errno, -fPIC
    name: gcc
    version: 15.2.0
  pythran:
    include directory: ../../../../../../../../software/SciPy-bundle/2026.05-gfbf-2026.1/lib/python3.14/site-packages/pythran
    version: 0.18.1
Machine Information:
  build:
    cpu: aarch64
    endian: little
    family: aarch64
    system: linux
  cross-compiled: false
  host:
    cpu: aarch64
    endian: little
    family: aarch64
    system: linux
Python Information:
  path: /tmp/software/Python/3.14.2-GCCcore-15.2.0/bin/python
  version: '3.14'
```

--- a/scipy/spatial/tests/test_distance.py
+++ b/scipy/spatial/tests/test_distance.py
@@ -790,7 +790,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-euclidean-iris']
         Y_test1 = wpdist_no_const(X, 'euclidean')
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.slow
     def test_pdist_euclidean_iris_float32(self):
@@ -798,7 +798,7 @@ class TestPdist:
         X = np.float32(eo['iris'])
         Y_right = eo['pdist-euclidean-iris']
         Y_test1 = wpdist_no_const(X, 'euclidean')
-        assert_allclose(Y_test1, Y_right, rtol=eps, verbose=verbose > 2)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7, verbose=verbose > 2)
 
     @pytest.mark.slow
     def test_pdist_euclidean_iris_nonC(self):
@@ -808,7 +808,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-euclidean-iris']
         Y_test2 = wpdist_no_const(X, 'test_euclidean')
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_seuclidean_random(self):
         eps = 1e-7
@@ -842,7 +842,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-seuclidean-iris']
         Y_test1 = pdist(X, 'seuclidean')
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_seuclidean_iris_float32(self):
         # Tests pdist(X, 'seuclidean') on the Iris data set (float32).
@@ -850,7 +850,7 @@ class TestPdist:
         X = np.float32(eo['iris'])
         Y_right = eo['pdist-seuclidean-iris']
         Y_test1 = pdist(X, 'seuclidean')
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7)
 
     def test_pdist_seuclidean_iris_nonC(self):
         # Test pdist(X, 'test_seuclidean') [the non-C implementation] on the
@@ -859,7 +859,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-seuclidean-iris']
         Y_test2 = pdist(X, 'test_seuclidean')
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_cosine_random(self):
         eps = 1e-7
@@ -943,7 +943,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-cityblock-iris']
         Y_test1 = wpdist_no_const(X, 'cityblock')
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.slow
     def test_pdist_cityblock_iris_float32(self):
@@ -951,7 +951,7 @@ class TestPdist:
         X = np.float32(eo['iris'])
         Y_right = eo['pdist-cityblock-iris']
         Y_test1 = wpdist_no_const(X, 'cityblock')
-        assert_allclose(Y_test1, Y_right, rtol=eps, verbose=verbose > 2)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7, verbose=verbose > 2)
 
     @pytest.mark.slow
     def test_pdist_cityblock_iris_nonC(self):
@@ -961,7 +961,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-cityblock-iris']
         Y_test2 = wpdist_no_const(X, 'test_cityblock')
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_correlation_random(self):
         eps = 1e-7
@@ -990,7 +990,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-correlation-iris']
         Y_test1 = wpdist(X, 'correlation')
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.slow
     def test_pdist_correlation_iris_float32(self):
@@ -998,7 +998,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = np.float32(eo['pdist-correlation-iris'])
         Y_test1 = wpdist(X, 'correlation')
-        assert_allclose(Y_test1, Y_right, rtol=eps, verbose=verbose > 2)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7, verbose=verbose > 2)
 
     @pytest.mark.slow
     def test_pdist_correlation_iris_nonC(self):
@@ -1009,7 +1009,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-correlation-iris']
         Y_test2 = wpdist(X, 'test_correlation')
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.parametrize("p", [0.1, 0.25, 1.0, 2.0, 3.2, np.inf])
     def test_pdist_minkowski_random_p(self, p):
@@ -1046,7 +1046,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-minkowski-3.2-iris']
         Y_test1 = wpdist_no_const(X, 'minkowski', p=3.2)
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.slow
     def test_pdist_minkowski_3_2_iris_float32(self):
@@ -1054,7 +1054,7 @@ class TestPdist:
         X = np.float32(eo['iris'])
         Y_right = eo['pdist-minkowski-3.2-iris']
         Y_test1 = wpdist_no_const(X, 'minkowski', p=3.2)
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7)
 
     @pytest.mark.slow
     def test_pdist_minkowski_3_2_iris_nonC(self):
@@ -1062,7 +1062,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-minkowski-3.2-iris']
         Y_test2 = wpdist_no_const(X, 'test_minkowski', p=3.2)
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.slow
     def test_pdist_minkowski_5_8_iris(self):
@@ -1070,7 +1070,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-minkowski-5.8-iris']
         Y_test1 = wpdist_no_const(X, 'minkowski', p=5.8)
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     @pytest.mark.slow
     def test_pdist_minkowski_5_8_iris_float32(self):
@@ -1078,7 +1078,7 @@ class TestPdist:
         X = np.float32(eo['iris'])
         Y_right = eo['pdist-minkowski-5.8-iris']
         Y_test1 = wpdist_no_const(X, 'minkowski', p=5.8)
-        assert_allclose(Y_test1, Y_right, rtol=eps, verbose=verbose > 2)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7, verbose=verbose > 2)
 
     @pytest.mark.slow
     def test_pdist_minkowski_5_8_iris_nonC(self):
@@ -1086,7 +1086,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-minkowski-5.8-iris']
         Y_test2 = wpdist_no_const(X, 'test_minkowski', p=5.8)
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_mahalanobis(self):
         # 1-dimensional observations
@@ -1192,7 +1192,7 @@ class TestPdist:
         X = eo['iris']
         Y_right = eo['pdist-jensenshannon-iris']
         Y_test2 = pdist(X, 'test_jensenshannon')
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_matching_mtica1(self):
         # Test matching(*,*) with mtica example #1 (nums).
@@ -2233,21 +2233,21 @@ class TestChebyshev:
         X = eo['iris']
         Y_right = eo['pdist-chebyshev-iris']
         Y_test1 = pdist(X, 'chebyshev')
-        assert_allclose(Y_test1, Y_right, rtol=eps)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=1e-15)
 
     def test_pdist_chebyshev_iris_float32(self):
         eps = 1e-5
         X = np.float32(eo['iris'])
         Y_right = eo['pdist-chebyshev-iris']
         Y_test1 = pdist(X, 'chebyshev')
-        assert_allclose(Y_test1, Y_right, rtol=eps, verbose=verbose > 2)
+        assert_allclose(Y_test1, Y_right, rtol=eps, atol=5e-7, verbose=verbose > 2)
 
     def test_pdist_chebyshev_iris_nonC(self):
         eps = 1e-14
         X = eo['iris']
         Y_right = eo['pdist-chebyshev-iris']
         Y_test2 = pdist(X, 'test_chebyshev')
-        assert_allclose(Y_test2, Y_right, rtol=eps)
+        assert_allclose(Y_test2, Y_right, rtol=eps, atol=1e-15)
 
     def test_weighted(self):
         # Basic test for weighted Chebyshev.  Only components with non-zero

